src/clients/archwayd/__test__/index.test.js

```diff
@@ -71,6 +71,44 @@ describe('DefaultArchwayClient', () => {
       expect(archwayd.called).toBeTruthy();
     });
   });
+
+  describe('runJson', () => {
+    beforeEach(() => {
+      spawk.clean();
+      spawk.preventUnmatched();
+    });
+
+    afterEach(() => {
+      spawk.done();
+      jest.clearAllMocks();
+    });
+
+    test('runs archwayd and parses the output as json', async () => {
+      const client = await createClient({ extraArgs: ['--keyring-backend', 'test'] });
+      const output = { txhash: '123456' };
+      const archwayd = spawk.spawn(client.command)
+        .stdout(JSON.stringify(output));
+
+      const json = await client.runJson('keys', ['list'], { printStdout: false });
+
+      expect(archwayd.calledWith).toMatchObject({
+        command: client.command,
+        args: ['--keyring-backend', 'test', 'keys', 'list', '--output', 'json'],
+        options: { stdio: 'pipe', maxBuffer: 1024 * 1024 }
+      });
+
+      expect(json).toMatchObject(output);
+    });
+
+    test('returns an empty JSON in case the command does not return a JSON line', async () => {
+      const client = await createClient({ extraArgs: ['--keyring-backend', 'test'] });
+      spawk.spawn(client.command);
+
+      const json = client.runJson('keys', ['list'], { printStdout: false });
+
+      expect(json).toMatchObject({});
+    });
+  });
```

Comment:
should we test a malformed JSON as well?

Reply:
Some commands don't output anything but are still valid. The test is just checking that the wrapper doesn't break.